lib/spack/llnl/util/filesystem.py

```diff
@@ -1091,15 +1091,35 @@ def visit_directory_tree(root, visitor, rel_path='', depth=0):
     for f in dir_entries:
         if sys.version_info >= (3, 5, 0):
             rel_child = os.path.join(rel_path, f.name)
-            islink, isdir = f.is_symlink(), f.is_dir()
+            islink = f.is_symlink()
+            # Windows dir entry issue with attempting to read
+            # looped symlinks results in too
+            # many reparse attempts and raises
+            # a CANT_RESOLVE_FILENAME error
+            # See https://bugs.python.org/msg357725
+            # for more context
+            try:
+                isdir = f.is_dir()
+            except OSError as e:
+                if is_windows and hasattr(e, 'winerror')\
+                   and e.winerror == 5 and islink:
+                    # if path is a symlink, determine destination and
+                    # evaluate file vs directory
+                    evald_link = os.readlink(f)
+                    full_path = os.path.join(dir, evald_link)\
```

Comment:
I don't see the variable `dir` defined elsewhere in this function so I'm surprised that the tests pass.

Reply:
It's defined on line 1083. `dir = os.path.join(root, rel_path)`

Reply:
Thanks for the pointer! I was definitely wrong here